browsergym/core/src/browsergym/core/env.py

```diff
@@ -277,9 +301,23 @@ def override_property(task, env, property):
             # goal is text + images
             elif isinstance(goal, dict):
                 goal_msg = goal["message"]
-                for image_url in goal["image_urls"]:
+                for image_i, image_url in enumerate(goal["image_urls"]):
                     self.chat.add_message(role="user_image", msg=image_url)
-
+                    if image_url.startswith("http"):
+                        image = Image.open(requests.get(image_url, stream=True).raw)
+                        image_base64 = pil_to_b64(image)
+                    elif image_url.startswith("data:image/png;base64,"):
+                        image_base64 = image_url
+                        image = b64_to_pil(image_base64)
+                    else:
+                        raise ValueError(f"Unexpected image_url: {image_url}")
+                    # save the image to a temporary (but persistent) PNG file
+                    if not os.path.exists("tmp"):
```

Comment:
This won't work when running agents in parallel, there will be name collisions.

Reply:
what do you think the workaround is? some benchmarks have the same template id across tasks as well, so using unique ids might cause name collisions as well